### PR TITLE
feat: foundation and initial content for es-ds-docs site

### PR DESCRIPTION
**Pull request by ericdouglaspratt**
### 🔗 Linked issue

- https://energysage.atlassian.net/browse/CED-1725

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Created components in `es-ds-components`
   - Created BContainer, BRow, and BCol
   - Created EsBreadcrumbs and EsCollapse to prove out examples of mapping ESDS 2.0 props API to PrimeVue components (also because the core docs site functionality needs them)
   - Was able to copy in IconChevronDown and IconChevronUp with no changes, to support EsCollapse and the `es-ds-docs` side nav, respectively
- Set up basic structure of `es-ds-docs` site
   - Installed PrimeVue and configured unstyled mode
   - Got the component/documentation source display working with prism
   - Proved out building the Colors page as an example of how we need to modify importing SASS variables in Vite
   - Copied over DsPropTable and associated DsResponsiveTable components

Screen Shot 2024-07-29 at 2 38 51 PM

### 🥼 Testing

- Tested locally with `make dev`

#### 🧐 Feedback Requested / Focus Areas

- Is this enough for people to get started building PrimeVue-based components?
- EsBreadcrumbs and EsCollapse have enough custom functionality (needed to preserve the props API from ESDS 2.0) that it's likely necessary to confine the pass-through styling to those specific components (as I have done) rather than specifying them globally - there may be cases where the PrimeVue component is solid enough that we want to specify the pass-through styles globally for that one, but we'll have to see. If we do, that'd likely be a separate example we'd want to call out on the docs page for that component.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

**Comment by hroth1994** [comment]
I ran `make install` and `make dev` and I'm getting some console errors:
`[4:00:20 PM]  ERROR  Internal server error: Failed to resolve import "@energysage/es-ds-components/components/es-collapse.vue?raw" from "pages/molecules/collapse.vue". Does the file exist?`
And a warning:
` WARN  [Vue warn]: Failed to resolve component: b-container
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.`

**Comment by mpleroux** [comment]
> I ran `make install` and `make dev` and I'm getting some console errors.

@hroth1994 I haven't gotten those errors.

I typically use the six commands listed in the README section [Installing Dependencies and Linking packages](https://github.com/EnergySage/es-ds?tab=readme-ov-file#installing-dependencies-and-linking-packages). Do you still see them if you try that?

**Comment by ericdouglaspratt** [comment]
@mpleroux I worked with Hannah yesterday to fix the commands that `make dev` was using to link the packages locally, and now it's working for her.

For the Vue 3 work, there are no more six commands - I've updated the README. You should only have to run `make install` and then `make dev`!

File: es-ds-docs/components/ds-code-block.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<template>
+    <!-- eslint-disable vue/no-v-html -->
+    <div class="prism">
```
Comment (mpleroux) [inline es-ds-docs/components/ds-code-block.vue:3]: A minor thing: The prism plugin doesn't seem to be kicking in for the code samples. I don't see line numbers or the Copy/View Source buttons when I hover over the code samples.

Comment (ericdouglaspratt) [reply es-ds-docs/components/ds-code-block.vue:3]: Ah good catch - I saw syntax highlighting and code and figured that was it.

I've created a ticket to fix Prism (https://energysage.atlassian.net/browse/CED-1764).

**Review by hroth1994 — APPROVED** [review]
The additional tickets created cover everything I've found. This LGTM 🌟

**Review by mpleroux — APPROVED** [review]

**Pull request merged**